data_processors/pipeline/services.py

```diff
@@ -361,46 +364,56 @@ def notify_batch_run_status(batch_run_id):
     # at the mo, BatchRun has only two states, RUNNING or not
     if batch_run.running:
         state = "running"
-        slack_color = libslack.SlackColor.BLUE.value
     else:
         state = "completed"
-        slack_color = libslack.SlackColor.GREEN.value
 
     workflows = Workflow.objects.get_by_batch_run(batch_run=batch_run)
     workflow = workflows[0]  # pick one for convenience
 
-    cnt = 0
-    stats = {
+    _total_cnt = 0
+    _stats = {
         WorkflowStatus.SUCCEEDED.value: 0,
         WorkflowStatus.FAILED.value: 0,
         WorkflowStatus.ABORTED.value: 0,
         WorkflowStatus.RUNNING.value: 0,
     }
-    mtx = ""
+    _metrics = ""
     for wfl in workflows:
-        mtx += f"{wfl.sample_name}: {str(wfl.end_status).upper()}, {wfl.wfr_id}\n"
-        cnt += 1
-        stats[wfl.end_status] += 1
+        _metrics += f"{wfl.sample_name}: {str(wfl.end_status).upper()}, {wfl.wfr_id}\n"
+        _total_cnt += 1
+        _stats[wfl.end_status] += 1
+
+    _title = f"Total: {_total_cnt} " \
+             f"| Running: {_stats[WorkflowStatus.RUNNING.value]} " \
+             f"| Succeeded: {_stats[WorkflowStatus.SUCCEEDED.value]} " \
+             f"| Failed: {_stats[WorkflowStatus.FAILED.value]} " \
+             f"| Aborted: {_stats[WorkflowStatus.ABORTED.value]}"
 
-    _title = f"Total: {cnt} " \
-             f"| Running: {stats[WorkflowStatus.RUNNING.value]} " \
-             f"| Succeeded: {stats[WorkflowStatus.SUCCEEDED.value]} " \
-             f"| Failed: {stats[WorkflowStatus.FAILED.value]} " \
-             f"| Aborted: {stats[WorkflowStatus.ABORTED.value]}"
+    _color = libslack.SlackColor.GRAY.value  # default to grey
+
+    if _total_cnt == _stats[WorkflowStatus.RUNNING.value] or _total_cnt == _stats[WorkflowStatus.SUCCEEDED.value]:
+        _color = libslack.SlackColor.GREEN.value
+
+    elif _total_cnt == _stats[WorkflowStatus.FAILED.value]:
+        _color = libslack.SlackColor.RED.value
+
+    elif _stats[WorkflowStatus.SUCCEEDED.value] > 0 and \
```

Comment:
This does not capture cases with only failed and aborted workflows, right?

Why not remove the condition altogether and have this as an `else` catch all?
This would give us:
- all running -> blue
- all succeeded -> green
- all failed -> red
- anything else -> orange

This only applies once all workflows have finished (one way or another), right?
Are there other states we'd expect?

Reply:
right, yep, sound good, will update it. yes, this is all workflows together, so, don't think any other more states there...